### DISCUSSION TITLE
Python integration with Pylint

### DISCUSSION THREAD

**Discussion by anshul-raman** [zed-industries/zed#14241] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

I want to integrate with external tools for python development, like isort and pylint. Is there any to integrate with Pylint. Additionally is there any way to run external commands on save like running isort.

**Comment by JosephTLyons** [comment]
Hey @anshul-raman, you should be able to get some of these tools configured to run by using tasks.

https://zed.dev/docs/tasks

We also just recently shipped the ability for extensions to provide tasks, so that extension authors can start making common things like this and shipping them for others to install - I'd expect to see those starting to pop up soon.

There's also a way you can configure custom formatters in Python - here's the documentation:

https://zed.dev/docs/languages/python#code-formatting

There is a limitation here though with the custom formatting, and that is that you can only currently configure one at a time:

- https://github.com/zed-industries/zed/issues/4822

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
